### DISCUSSION TITLE
Time to NodalValueSampler

### DISCUSSION THREAD

**Discussion by agniksr** [idaholab/moose#27944] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

So i am trying to output the temperature at certain nodes in my mesh file as below, for n timesteps. To get all the data in one file, i used contains_complete_history = true.  I want to plot the Temperature vs time for 4 or 5 radii (x^2+y^2), for a fixed z (= 0 8 16 24 32 40). 
However, when i open the csv file, the same node id is not present 10 times, for instance the node id 2 is only present once. Is the node id updated after every iteration?
Any idea how i can obtain the time along with the other vpp values, so that it is easier to identify the state of the nodes at a certain timestep

```
[VectorPostprocessors]
   [./AllElementTemp]
     type = NodalValueSampler
     variable = 'T'
     sort_by = id
     contains_complete_history = true
     use_displaced_mesh = true
     boundary = 'bottom z8 z16 z24 z32 top'
   [../]
[]
```

**Comment by GiudGiud** [comment]
>  Is the node id updated after every iteration?

This is not the case by default, but you are using displaced meshes so that could happen if repartitioning and consequently re-numbering happened in parallel.

>      contains_complete_history = true

Using different files, one for each time step, is less messy than this option. 
Especially sorting by id rather than than time, I don't know what the output will be ordered like

**Reply by agniksr** [reply]
Okay thank you for the answer. I changed it to false and made a script file to sort through the multiple files.

**Comment by jessecarterMOOSE** [comment]
@GiudGiud Is there a way to get time to be output to the VPPfile?

**Reply by GiudGiud** [reply]
not that I can see. 
the timestep is in the name of the csv
otherwise you can add time as an auxvariable

```
[AuxVariables/time]
[]
[AuxKernels/set_time]
  type = ParsedAux
  expression = 't'
use_xyzt = true
[]
```

**Reply by jessecarterMOOSE** [reply]
Yeah I'm after _time_ not _timestep._ The AuxVariable worked :+1:
